src/Voron/Impl/FreeSpace/FreeSpaceHandling.cs

```diff
@@ -192,19 +194,31 @@ public void Clear()
             {
                 var current = new StreamBitArray(it.CreateReaderForCurrent());
 
+                long currentSectionId = it.CurrentKey;
                 long? page;
-                if (current.SetCount < num)
+                if (_maxConsecutiveRangePerSection.TryGetValue(currentSectionId, out var knownMax) && knownMax <= num)
                 {
-                    if (TryFindSmallValueMergingTwoSections(tx, freeSpaceTree, it.CurrentKey, num, current, out page))
+                    // current section's maximum continuous range is insufficient for the requested size.
+                    // while we can skip searching within this section alone, we still need to check
+                    // if merging with the following section could provide enough space
+                    if (TryFindSmallValueMergingTwoSections(tx, freeSpaceTree, currentSectionId, num, current, out page))
```

Comment:
The issue is that this requires us to do work _per section_.
I agree that this is useful, but let's change the way we do that. In the `_maxConsecutiveRangePerSection`, store (in one int) three numbers (maxBits, set bits at start, set bits at end).

That way, we don't have to touch any date.

Reply:
done